field/tests/helpers_test.rs

```diff
@@ -0,0 +1,304 @@
+mod helpers {
+    use p3_baby_bear::BabyBear;
+    use p3_field::{
+        PrimeCharacteristicRing, PrimeField32, PrimeField64, add_scaled_slice_in_place,
+        dot_product, field_to_array, halve_u32, halve_u64, reduce_32, scale_vec, split_32,
+        two_adic_coset_vanishing_polynomial, two_adic_subgroup_vanishing_polynomial,
+    };
+
+    #[test]
+    fn test_two_adic_subgroup_vanishing_polynomial() {
+        // x = 3, log_n = 3 → compute x^8 - 1
+        let x = BabyBear::from_u64(3);
+        let log_n = 3;
+
+        // x^8 = 3^8
+        let x_pow = x * x * x * x * x * x * x * x;
```

Comment:
Another option here would be: (But I'm fine with the current one) 
```suggestion
        // x^8 = 3^8 = 6561
        let x_pow = BabyBear::from_u16(6561);
```